astroplan/core.py

```diff
@@ -22,109 +38,152 @@
+            The local timezone, as either an instance of `pytz.timezone()` or
+            the string accepted by `pytz.timezone()`.
 
-        description : str
+        description : str (optional)
             A short description of the telescope, observatory or observing
             location.
         """
-        raise NotImplementedError()
-
-    def set_environment(self, pressure, relative_humidity, temperature):
-        """
-        Updates the Observer object with environmental conditions.
 
-        <longer description>
+        if elevation is None:
+            elevation = 0*u.m
 
-        Parameters
-        ----------
-        pressure : `~astropy.units.Quantity`
-            The ambient pressure.
+        self.name = name
+        self.pressure = pressure
+        self.temperature = temperature
+        self.relative_humidity = relative_humidity
 
-        relative_humidity : `~astropy.units.Quantity`
-            The ambient relative humidity.
+        # If lat/long given instead of EarthLocation, convert them
+        # to EarthLocation
+        if location is None and (latitude is not None and
+                                 longitude is not None):
+            self.location = EarthLocation.from_geodetic(longitude, latitude,
+                                                        elevation)
 
-        temperature :
-            The ambient temperature.
-        """
-        raise NotImplementedError()
+        elif isinstance(location, EarthLocation):
+            self.location = location
 
-    def get_date(self, input_date_time, timezone=None):
-        """
-        Builds an object containing date and time information.
+        else:
+            raise TypeError('Observatory location must be specified with '
+                            'either (1) an instance of '
+                            'astropy.coordinates.EarthLocation or (2) '
+                            'latitude and longitude in degrees as '
+                            'accepted by astropy.coordinates.Latitude and '
+                            'astropy.coordinates.Latitude.')
 
-        Default time zone is UTC.
-        If time zone (local or otherwise) requested, then uses `datetime.datetime`
-        and pytz.timezone to convert from UTC.
+        # Accept various timezone inputs, default to UTC
+        if isinstance(timezone, datetime.tzinfo):
+            self.timezone = timezone
+        elif isinstance(timezone, str) or isinstance(timezone, unicode):
+            self.timezone = pytz.timezone(timezone)
+        else:
+            raise TypeError('timezone keyword should be a string, or an '
+                            'instance of datetime.tzinfo')
 
-        Must first create date_time object with get_date() before using any other 
-        Observer methods, all of which require one as an argument.
+    def altaz(self, time, target=None, obswl=None):
+        """
+        Calculates an altitude/azimuth frame (if `target` not specified)
```

Comment:
"calculates" -> "generates"?  without a target no actual "calcualtion" is done in the sense of a coordinate transform.  So maybe something like ``If `target` is None, generates an altitude/azimuth frame.  Otherwise, calculates the transformation to that frame for the requested `target`."`` ?